From 7237e122fbc678d59621ccf5cf127e47930349ed Mon Sep 17 00:00:00 2001
From: Aaron Williams <aaron.williams@cavium.com>
Date: Tue, 28 Jun 2016 15:27:24 -0700
Subject: [PATCH 0510/1239] Refactored bootloader code to put files in the
 correct locations

A lot of changes to the Kconfig configuration were made in order to support
moving files to the correct locations.  Note that 88xx has not yet been tested
and is now likely broken.

Signed-off-by: Aaron Williams <aaron.williams@cavium.com>
---
 arch/arm/Kconfig                       |  10 +-
 arch/arm/Makefile                      |   2 +
 arch/arm/mach-thunderx/Kconfig         | 173 +++++++++++++++++++++++++
 arch/arm/mach-thunderx/Makefile        |  10 ++
 arch/arm/mach-thunderx/lowlevel_init.S |  71 ++++++++++
 board/cavium/common/env_atf.c          |   6 +-
 board/cavium/thunderx/Kconfig          |  55 ++++++--
 board/cavium/thunderx/atf.c            |   6 +-
 board/cavium/thunderx/cmd_vnic.c       |   2 +-
 board/cavium/thunderx/fdt.c            |   7 +-
 board/cavium/thunderx/psci.c           |   2 +-
 board/cavium/thunderx/thunderx.c       |  14 +-
 configs/thunderx_81xx_defconfig        |  13 ++
 configs/thunderx_88xx_defconfig        |   3 +
 drivers/i2c/Kconfig                    |   7 +
 drivers/net/cavium/nic_main.c          |   2 +-
 drivers/net/cavium/nicvf_main.c        |   2 +-
 drivers/net/cavium/thunder_bgx.c       |   4 +-
 drivers/net/cavium/thunder_xcv.c       |  18 +--
 include/cavium/slt.h                   |  11 --
 include/cavium/thunderx.h              |  13 --
 include/cavium/thunderx_fdt.h          |  11 --
 include/cavium/thunderx_smi.h          |  12 --
 include/cavium/thunderx_vnic.h         |  20 ---
 include/configs/thunderx_81xx.h        |  56 --------
 25 files changed, 362 insertions(+), 168 deletions(-)
 create mode 100644 arch/arm/mach-thunderx/Kconfig
 create mode 100644 arch/arm/mach-thunderx/Makefile
 create mode 100644 arch/arm/mach-thunderx/lowlevel_init.S
 delete mode 100644 include/cavium/slt.h
 delete mode 100644 include/cavium/thunderx.h
 delete mode 100644 include/cavium/thunderx_fdt.h
 delete mode 100644 include/cavium/thunderx_smi.h
 delete mode 100644 include/cavium/thunderx_vnic.h

diff --git a/arch/arm/Kconfig b/arch/arm/Kconfig
index f3c20cc6eb..c1f57207f3 100644
--- a/arch/arm/Kconfig
+++ b/arch/arm/Kconfig
@@ -390,9 +390,6 @@ config TPL_SYS_THUMB_BUILD
 	   density. For ARM architectures that support Thumb2 this flag will
 	   result in Thumb2 code generated by GCC.
 
-config THUNDERX
-	bool
-
 config SYS_L2CACHE_OFF
 	bool "L2cache off"
 	help
@@ -1614,8 +1611,8 @@ config ARCH_ASPEED
 	select OF_CONTROL
 	imply CMD_DM
 
-config TARGET_THUNDERX_81XX
-	bool "Support ThunderX 81xx"
+config ARCH_THUNDERX
+	bool "Support Cavium ThunderX SoCs"
 	select THUNDERX
 	select OF_CONTROL
 	select DM
@@ -1623,6 +1620,7 @@ config TARGET_THUNDERX_81XX
 	select DM_PCI
 	select DM_PCI_COMPAT
 	select OF_SEPARATE
+	select ARM64
 
 endchoice
 
@@ -1694,6 +1692,8 @@ source "arch/arm/mach-imx/mx2/Kconfig"
 
 source "arch/arm/mach-imx/mx3/Kconfig"
 
+source "arch/arm/mach-thunderx/Kconfig"
+
 source "arch/arm/mach-imx/mx5/Kconfig"
 
 source "arch/arm/mach-imx/mx6/Kconfig"
diff --git a/arch/arm/Makefile b/arch/arm/Makefile
index 5384981c17..bff453ac41 100644
--- a/arch/arm/Makefile
+++ b/arch/arm/Makefile
@@ -43,6 +43,7 @@ tune-$(CONFIG_CPU_ARM1176)	=
 tune-$(CONFIG_CPU_V7A)		=-mtune=generic-armv7-a
 tune-$(CONFIG_CPU_V7R)		=
 tune-$(CONFIG_ARM64)		=
+tune-$(CONFIG_THUNDERX)		=
 
 # Evaluate tune cc-option calls now
 tune-y := $(tune-y)
@@ -79,6 +80,7 @@ machine-$(CONFIG_ARCH_ROCKCHIP)		+= rockchip
 machine-$(CONFIG_STM32)			+= stm32
 machine-$(CONFIG_ARCH_STM32MP)		+= stm32mp
 machine-$(CONFIG_TEGRA)			+= tegra
+machine-$(CONFIG_ARCH_THUNDERX)		+= thunderx
 machine-$(CONFIG_ARCH_UNIPHIER)		+= uniphier
 machine-$(CONFIG_ARCH_ZYNQ)		+= zynq
 machine-$(CONFIG_ARCH_ZYNQMP)		+= zynqmp
diff --git a/arch/arm/mach-thunderx/Kconfig b/arch/arm/mach-thunderx/Kconfig
new file mode 100644
index 0000000000..ce997e5e41
--- /dev/null
+++ b/arch/arm/mach-thunderx/Kconfig
@@ -0,0 +1,173 @@
+if ARCH_THUNDERX
+
+config	THUNDERX_COMMON
+	bool
+	default y
+	select DM
+	select DM_I2C
+	select DM_I2C_COMPAT
+	select DM_GPIO
+	select DM_PCI
+	select DM_PCI_COMPAT
+	select DM_SERIAL
+	select DM_USB
+	select OF_CONTROL
+	select OF_EMBED
+	select OF_LIBFDT
+	select ARM64
+	select PCI_THUNDERX_ECAM
+	select PL01X_SERIAL
+	select SYS_64BIT
+	select NO_RELOCATION
+	select PCI
+	select SYS_PCI_64BIT
+	select PCI_PNP
+	select PCI_AUTO
+
+choice
+	prompt "Choose target processor"
+
+	config TARGET_THUNDERX_CN81XX
+		bool "ThunderX CN81XX family"
+		select SYS_SOC_THUNDERX_81XX
+
+	config TARGET_THUNDERX_CN88XX
+		bool "ThunderX CN88XX family"
+		select SYS_SOC_THUNDERX_88XX
+
+endchoice
+
+config PCI_AUTO
+	bool
+	default y
+
+config OF_LIBFDT
+	bool
+	default y
+
+config SYS_ARCH
+	string
+	default "thunderx"
+
+config SYS_VENDOR
+	string
+	default	"cavium"
+
+config SYS_SOC
+	string
+	default "thunderx"
+
+config SYS_CPU
+	string
+	default "armv8"
+
+config PL01X_SERIAL
+	bool
+	default y
+
+config PL011_CLOCK
+	int
+	default 24000000
+	help
+	 Default clock for the built-in serial port.
+
+config SYS_SERIAL0
+	hex
+	default 0x87e028000000
+
+config SYS_SERIAL1
+	hex
+	default 0x87e029000000
+
+config AP_STACK_ALIGN
+	int
+	default 16
+	help
+	 Stack alignment for PSCI calls
+
+config AP_STACK_SIZE
+	hex
+	default 10000
+	help
+	 Stack size for PSCI calls
+
+config SYS_64BIT
+	bool
+	default y
+
+config SYS_VA_BITS
+	int
+	default 48
+
+config SYS_PTL2_BITS
+	int
+	default 42
+
+config SYS_BLOCK_SHIFT
+	int
+	default 29
+
+config SYS_PTL1_ENTRIES
+	int
+	default 64
+
+config SYS_PTL2_ENTRIES
+	int
+	default 8192
+
+config NO_RELOCATION
+	bool
+	default y
+
+config PCI
+	bool
+	default y
+
+config SYS_PCI_64BIT
+	bool
+	default y
+
+config PCI_PNP
+	bool
+	default y
+
+config SYS_PCI_ARI
+	bool
+	default y
+
+config DM_I2C
+	bool
+	default y
+
+config DM_I2C_COMPAT
+	bool
+	default y
+
+config DM_GPIO
+	bool
+	default y
+
+config CMD_ATF
+	bool "Enable ATF query commands"
+	default y
+	help
+	 Enable vendor specific ATF query commands such as SPI and SD/MMC
+	 devices access, low level environment query, boot device layout
+	 and node count.
+
+config SYS_CACHELINE_SIZE
+	int
+	default 128
+
+config PCI_THUNDERX_ECAM
+	bool
+	default y
+	help
+	 This enables support for the PCI Common Configuration Unit (PCC) in
+	 the ThunderX series of processors.  The PCC implements the PCI
+	 Enhanced Configuration Access Mechanism (ECAM), and PCI configuration
+	 blocks for ThunderX on-chip functions.
+
+#source "arch/arm/mach-octeontx/cn81xx/Kconfig"
+
+endif
diff --git a/arch/arm/mach-thunderx/Makefile b/arch/arm/mach-thunderx/Makefile
new file mode 100644
index 0000000000..cebcddb5f9
--- /dev/null
+++ b/arch/arm/mach-thunderx/Makefile
@@ -0,0 +1,10 @@
+#
+# Copyright 2014-2016, Cavium Networks, Inc
+#
+# SPDX-License-Identifier:	GPL-2.0+
+#
+
+obj-y += lowlevel_init.o
+
+obj-$(CONFIG_THUNDERX_CN81XX)	+= cn81xx/
+obj-$(CONFIG_THUNDERX_CN88XX)	+= cn88xx/
diff --git a/arch/arm/mach-thunderx/lowlevel_init.S b/arch/arm/mach-thunderx/lowlevel_init.S
new file mode 100644
index 0000000000..33fb0b14d9
--- /dev/null
+++ b/arch/arm/mach-thunderx/lowlevel_init.S
@@ -0,0 +1,71 @@
+/*
+ * (C) Copyright 2014-2016, Cavium Networks, Inc
+ *
+ * SPDX-License-Identifier:	GPL-2.0+
+ *
+ * Extracted from armv8/start.S
+ */
+
+#include <config.h>
+#include <linux/linkage.h>
+#include <asm/arch/thunderx_svc.h>
+
+#include <asm/macro.h>
+
+ENTRY(lowlevel_init)
+	mov	x29, lr			/* Save LR */
+
+	ldr	x0, =THUNDERX_FDT_GET       /* Copy the FDT to end of u-boot binary*/
+	adr	x1, _end
+	ldr	x2, =CONFIG_BDK_FDT_SIZE
+	smc	#0
+
+	ldr	x0, =THUNDERX_DRAM_SIZE
+	mov	x1, xzr
+	smc	#0			/* Get the DRAM size */
+
+	mov	x23, #2			/* Select L2 */
+	msr	csselr_el1, x23
+	mrs	x23, ccsidr_el1
+	ubfx	x20, x23, #13, #15	/* x20 = Sets - 1 */
+	ubfx	x21, x23, #3, #10	/* x21 = Assoc(Ways) - 1 */
+
+	ubfx	x22, x23, #0, #2
+	add	x22, x22, #4
+	mov	x23, #1
+	lsl	x22, x23, x22		/* x22 = Line Size */
+
+	add	x20, x20, #1		/* x20 = Sets */
+	add	x21, x21, #1		/* x21 = Ways */
+
+	mul	x20, x20, x21		/* x20 = Sets * Ways */
+	mul	x20, x20, x22		/* x20 = L2size */
+
+	sub	x21, x0, x20		/* x21 = last address - L2size */
+
+	mrs	x23, dczid_el0
+	ubfx	x1, x23, #4, #1		/* x1 = DZP bit */
+	cbnz	x1, 3f			/* If dc zva is prohibited, do nothing */
+	and	x23, x23, #15
+	add	x23, x23, #2
+	mov	x24, #1
+	lsl	x23, x24, x23		/* ZVA block size */
+
+1:
+	dc	zva, x21		/* Zero the end of DRAM block by block */
+	add	x21, x21, x23
+	cmp	x21, x0
+	b.lt	1b
+
+	sub	x21, x0, x20		/* x21 = last address - L2size */
+2:
+	sys	#0,c11,C1,#2, x21	/* Flush and invalidate L2 line by line */
+	add	x21, x21, x22
+	cmp	x21, x0
+	b.lt	2b
+
+3:
+	mov	lr, x29			/* Restore LR */
+	ret
+ENDPROC(lowlevel_init)
+
diff --git a/board/cavium/common/env_atf.c b/board/cavium/common/env_atf.c
index d845f031ab..23cf0d3134 100755
--- a/board/cavium/common/env_atf.c
+++ b/board/cavium/common/env_atf.c
@@ -20,9 +20,9 @@
 #include <linux/stddef.h>
 #include <memalign.h>
 
-#include <cavium/atf.h>
-#include <cavium/atf_part.h>
-#include <cavium/thunderx_svc.h>
+#include <asm/arch/atf.h>
+#include <asm/arch/atf_part.h>
+#include <asm/arch/thunderx_svc.h>
 
 #include <search.h>
 #include <errno.h>
diff --git a/board/cavium/thunderx/Kconfig b/board/cavium/thunderx/Kconfig
index 7187661519..4f113d7b1f 100644
--- a/board/cavium/thunderx/Kconfig
+++ b/board/cavium/thunderx/Kconfig
@@ -1,16 +1,32 @@
-if TARGET_THUNDERX_88XX || TARGET_THUNDERX_81XX
 
-config SYS_CPU
-	string
-	default "armv8"
+choice
+	prompt "Target board"
+
+config	TARGET_THUNDERX_81XX
+	bool "ThunderX CN81XX"
+
+config	TARGET_THUNDERX_88XX
+	bool "ThunderX CN88XX"
+
+endchoice
 
 config SYS_BOARD
 	string
 	default "thunderx"
 
-config SYS_VENDOR
-	string
-	default "cavium"
+config DM_I2C
+	bool
+	default y
+
+config DM_I2C_COMPAT
+	bool
+	default y
+
+config DM_GPIO
+	bool
+	default y
+
+if TARGET_THUNDERX_88XX || TARGET_THUNDERX_81XX
 
 config CMD_ATF
 	bool "Enable ATF query commands"
@@ -19,21 +35,32 @@ config CMD_ATF
 	  Enable vendor specific ATF query commands such as SPI and SD/MMC
 	  devices access, low level environment query, boot device layout
 	  and node count.
-
 endif
 
 if TARGET_THUNDERX_88XX
 
-config SYS_CONFIG_NAME
-	string
-	default "thunderx_88xx"
+	config SYS_CONFIG_NAME
+		string
+		default "thunderx_88xx"
+
+	config SYS_PROMPT
+		string "Prompt to use"
+		default	"ThunderX_88XX> "
 
 endif
 
 if TARGET_THUNDERX_81XX
 
-config SYS_CONFIG_NAME
-	string
-	default "thunderx_81xx"
+	config SYS_CONFIG_NAME
+		string
+		default "thunderx_81xx"
+
+	config IDENT_STRING
+		string
+		default " for Cavium OcteonTX CN81XX ARM V8 Core"
+
+	config SYS_PROMPT
+		string "Prompt to use"
+		default	"ThunderX_81XX> "
 
 endif
diff --git a/board/cavium/thunderx/atf.c b/board/cavium/thunderx/atf.c
index 4c47ba9a6f..d9de91de84 100644
--- a/board/cavium/thunderx/atf.c
+++ b/board/cavium/thunderx/atf.c
@@ -7,9 +7,9 @@
 #include <asm/io.h>
 
 #include <asm/system.h>
-#include <cavium/thunderx_svc.h>
-#include <cavium/atf.h>
-#include <cavium/atf_part.h>
+#include <asm/arch/thunderx_svc.h>
+#include <asm/arch/atf.h>
+#include <asm/arch/atf_part.h>
 
 #include <asm/psci.h>
 
diff --git a/board/cavium/thunderx/cmd_vnic.c b/board/cavium/thunderx/cmd_vnic.c
index 504d1ccf05..91d2526cb8 100644
--- a/board/cavium/thunderx/cmd_vnic.c
+++ b/board/cavium/thunderx/cmd_vnic.c
@@ -16,7 +16,7 @@
 #include <net.h>
 
 #include <malloc.h>
-#include <cavium/atf.h>
+#include <asm/arch/atf.h>
 
 #include <libfdt.h>
 
diff --git a/board/cavium/thunderx/fdt.c b/board/cavium/thunderx/fdt.c
index 03a0c2a73b..ddd4d0eb5e 100644
--- a/board/cavium/thunderx/fdt.c
+++ b/board/cavium/thunderx/fdt.c
@@ -3,7 +3,7 @@
  *
  * SPDX-License-Identifier:	GPL-2.0+
 **/
-
+#define DEBUG
 #include <common.h>
 #include <malloc.h>
 #include <errno.h>
@@ -12,10 +12,10 @@
 #include <libfdt.h>
 #include <fdtdec.h>
 #include <fdt_support.h>
-#include <cavium/atf.h>
+#include <asm/arch/atf.h>
 
 #ifdef CONFIG_THUNDERX_VNIC
-# include <cavium/thunderx_vnic.h>
+# include <asm/arch/thunderx_vnic.h>
 #endif
 
 #define MAX_LMAC_PER_BGX 4
@@ -66,6 +66,7 @@ void thunderx_parse_bdk_config(void)
 		return;
 	}
 
+	debug("%s: fdt blob at %p\n", __func__, gd->fdt_blob);
 	ret = fdt_check_header(gd->fdt_blob);
 	if (ret < 0) {
 		printf("fdt: %s\n", fdt_strerror(ret));
diff --git a/board/cavium/thunderx/psci.c b/board/cavium/thunderx/psci.c
index bbbb876d9e..5e9ccb43cd 100644
--- a/board/cavium/thunderx/psci.c
+++ b/board/cavium/thunderx/psci.c
@@ -12,7 +12,7 @@
 #include <asm/bootm.h>
 #include <asm/armv8/mmu.h>
 
-#include <cavium/slt.h>
+#include <asm/arch/slt.h>
 
 #ifdef CONFIG_OF_LIBFDT
 #include <libfdt.h>
diff --git a/board/cavium/thunderx/thunderx.c b/board/cavium/thunderx/thunderx.c
index 79933a31ab..b1e8080052 100644
--- a/board/cavium/thunderx/thunderx.c
+++ b/board/cavium/thunderx/thunderx.c
@@ -2,7 +2,7 @@
 /**
  * (C) Copyright 2014, Cavium Inc.
 **/
-
+#define DEBUG
 #include <common.h>
 #include <dm.h>
 #include <malloc.h>
@@ -12,7 +12,7 @@
 
 #include <linux/compiler.h>
 
-#include <cavium/atf.h>
+#include <asm/arch/atf.h>
 
 #include <libfdt.h>
 #include <fdt_support.h>
@@ -47,6 +47,11 @@ U_BOOT_DEVICE(thunderx_serial1) = {
 	.platdata = &serial1,
 };
 #endif
+#include <asm/arch/thunderx_fdt.h>
+#include <asm/arch/atf.h>
+#include <dm/util.h>
+
+#include "cavm-arch.h"
 
 DECLARE_GLOBAL_DATA_PTR;
 
@@ -106,10 +111,15 @@ void reset_cpu(ulong addr)
  */
 int board_late_init(void)
 {
+	debug("%s()\n", __func__);
 	thunderx_parse_bdk_config();
 
 	printf("Board type: %s\n", getenv("board"));
 
+#ifdef DEBUG
+	dm_dump_all();
+#endif
+
 	return 0;
 }
 
diff --git a/configs/thunderx_81xx_defconfig b/configs/thunderx_81xx_defconfig
index aa97947a91..f247744a60 100644
--- a/configs/thunderx_81xx_defconfig
+++ b/configs/thunderx_81xx_defconfig
@@ -1,4 +1,7 @@
 CONFIG_ARM=y
+CONFIG_THUNDERX_COMMON=y
+#CONFIG_SYS_SOC="thunderx"
+CONFIG_ARCH_THUNDERX=y
 CONFIG_TARGET_THUNDERX_81XX=y
 CONFIG_DM_I2C=y
 CONFIG_DM_GPIO=y
@@ -27,6 +30,8 @@ CONFIG_DM_I2C_COMPAT=y
 CONFIG_MISC=y
 CONFIG_PHYLIB=y
 CONFIG_NETDEVICES=y
+CONFIG_PCI=y
+CONFIG_PCI_PNP=y
 CONFIG_PCI_THUNDERX_ECAM=y
 CONFIG_DEBUG_UART=y
 CONFIG_DEBUG_UART_PL011=y
@@ -37,3 +42,11 @@ CONFIG_USB=y
 CONFIG_DM_USB=y
 CONFIG_USB_XHCI_HCD=y
 CONFIG_USB_STORAGE=y
+CONFIG_SYS_CONFIG_NAME="thunder_81xx"
+CONFIG_DM_RTC=y
+CONFIG_SHA1=y
+CONFIG_SHA256=y
+CONFIG_ERRNO_STR=y
+CONFIG_SYS_I2C_THUNDERX_SLAVE_ADDR=0x77
+CONFIG_ENV_VARS_UBOOT_CONFIG=y
+CONFIG_OF_LIBFDT=y
diff --git a/configs/thunderx_88xx_defconfig b/configs/thunderx_88xx_defconfig
index 4254d3d6a4..338012aff0 100644
--- a/configs/thunderx_88xx_defconfig
+++ b/configs/thunderx_88xx_defconfig
@@ -1,4 +1,5 @@
 CONFIG_ARM=y
+CONFIG_ARCH_THUNDERX=y
 CONFIG_TARGET_THUNDERX_88XX=y
 CONFIG_SYS_TEXT_BASE=0x00500000
 CONFIG_NR_DRAM_BANKS=1
@@ -48,3 +49,5 @@ CONFIG_PCI_THUNDERX_ECAM=y
 CONFIG_DEBUG_UART=y
 CONFIG_DEBUG_UART_PL011=y
 CONFIG_DEBUG_UART_SKIP_INIT=y
+CONFIG_SYS_I2C_SPEED=100000
+CONFIG_SYS_I2C_SLAVE=0x77
diff --git a/drivers/i2c/Kconfig b/drivers/i2c/Kconfig
index 03d2fed341..384a9051c7 100644
--- a/drivers/i2c/Kconfig
+++ b/drivers/i2c/Kconfig
@@ -398,6 +398,13 @@ config SYS_I2C_TEGRA
 	help
 	  Support for NVIDIA I2C controller available in Tegra SoCs.
 
+config SYS_I2C_THUNDERX
+	bool "ThunderX i2c driver"
+	depends on ARCH_THUNDERX && DM_I2C
+	default y
+	help
+	 Enable I2C support for Cavium ThunderX line of processors.
+
 config SYS_I2C_UNIPHIER
 	bool "UniPhier I2C driver"
 	depends on ARCH_UNIPHIER && DM_I2C
diff --git a/drivers/net/cavium/nic_main.c b/drivers/net/cavium/nic_main.c
index 4989d32365..2d87973b2e 100644
--- a/drivers/net/cavium/nic_main.c
+++ b/drivers/net/cavium/nic_main.c
@@ -18,7 +18,7 @@
 #include <pci.h>
 #include <asm/io.h>
 
-#include <cavium/thunderx_vnic.h>
+#include <asm/arch/thunderx_vnic.h>
 
 #include "nic_reg.h"
 #include "nic.h"
diff --git a/drivers/net/cavium/nicvf_main.c b/drivers/net/cavium/nicvf_main.c
index 30fde0e046..71421a7aef 100644
--- a/drivers/net/cavium/nicvf_main.c
+++ b/drivers/net/cavium/nicvf_main.c
@@ -17,7 +17,7 @@
 #include <malloc.h>
 #include <asm/io.h>
 
-#include <cavium/thunderx_vnic.h>
+#include <asm/arch/thunderx_vnic.h>
 
 #include "nic_reg.h"
 #include "nic.h"
diff --git a/drivers/net/cavium/thunder_bgx.c b/drivers/net/cavium/thunder_bgx.c
index 33f0c3b237..d85eef6f29 100644
--- a/drivers/net/cavium/thunder_bgx.c
+++ b/drivers/net/cavium/thunder_bgx.c
@@ -24,8 +24,8 @@
  #include <fdt_support.h>
 #endif
 
-#include <cavium/thunderx_smi.h>
-#include <cavium/thunderx_vnic.h>
+#include <asm/arch/thunderx_smi.h>
+#include <asm/arch/thunderx_vnic.h>
 
 #include "nic_reg.h"
 #include "nic.h"
diff --git a/drivers/net/cavium/thunder_xcv.c b/drivers/net/cavium/thunder_xcv.c
index 548851d074..535e69a610 100644
--- a/drivers/net/cavium/thunder_xcv.c
+++ b/drivers/net/cavium/thunder_xcv.c
@@ -24,8 +24,8 @@
  #include <fdt_support.h>
 #endif
 
-#include <cavium/thunderx_smi.h>
-#include <cavium/thunderx_vnic.h>
+#include <asm/arch/thunderx_smi.h>
+#include <asm/arch/thunderx_vnic.h>
 
 #include "cavm-arch.h"
 #define	PCI_DEVICE_ID_THUNDER_XCV	0xA056
@@ -108,8 +108,8 @@ int xcv_setup_link(bool link_up, int link_speed)
 		   - XCV will continue to return TX credits for each tick
 		     that is sent on the TX data path */
 		reset.u = readq(CSR_PA(0, XCVX_RESET(0)));
-		reset.s.tx_dat_rst_n = 1; 
-		reset.s.rx_dat_rst_n = 1; 
+		reset.s.tx_dat_rst_n = 1;
+		reset.s.rx_dat_rst_n = 1;
 		writeq(reset.u, CSR_PA(0, XCVX_RESET(0)));
 	}
 
@@ -118,8 +118,8 @@ int xcv_setup_link(bool link_up, int link_speed)
 	   - XCV will continue to return TX credits for each tick that is sent
 	     on the TX data path */
 	reset.u = readq(CSR_PA(0, XCVX_RESET(0)));
-	reset.s.tx_dat_rst_n = link_up; 
-	reset.s.rx_dat_rst_n = link_up; 
+	reset.s.tx_dat_rst_n = link_up;
+	reset.s.rx_dat_rst_n = link_up;
 	writeq(reset.u, CSR_PA(0, XCVX_RESET(0)));
 
 	/* Full reset when link is down */
@@ -162,14 +162,14 @@ static const struct udevice_id thunderx_xcv_ids[] = {
 	{}
 };
 
-U_BOOT_DRIVER(thunderx_xcv) = {                         
+U_BOOT_DRIVER(thunderx_xcv) = {
         .name   = "thunderx_xcv",
         .id     = UCLASS_MISC,
-        .probe  = thunderx_xcv_probe,                   
+        .probe  = thunderx_xcv_probe,
         .of_match = thunderx_xcv_ids,
         .ops    = &thunderx_xcv_ops,
         .priv_auto_alloc_size = sizeof(struct lxcv),
-};                                                      
+};
 
 static struct pci_device_id thunderx_xcv_supported[] = {
         { PCI_VDEVICE(CAVIUM, PCI_DEVICE_ID_THUNDER_XCV) },
diff --git a/include/cavium/slt.h b/include/cavium/slt.h
deleted file mode 100644
index 8ef89ba17c..0000000000
--- a/include/cavium/slt.h
+++ /dev/null
@@ -1,11 +0,0 @@
-/**
- * (C) Copyright 2014, Cavium Inc.
- *
- * SPDX-License-Identifier:	GPL-2.0+
-**/
-#ifndef _SLT_H_
-#define _SLT_H_
-
-unsigned long slt_main(void);
-
-#endif /* _SLT_H_ */
diff --git a/include/cavium/thunderx.h b/include/cavium/thunderx.h
deleted file mode 100644
index 2fb02af075..0000000000
--- a/include/cavium/thunderx.h
+++ /dev/null
@@ -1,13 +0,0 @@
-/**
- * (C) Copyright 2014, Cavium Inc.
- *
- * SPDX-License-Identifier:	GPL-2.0+
-**/
-#ifndef __THUNDERX_H__
-#define __THUNDERX_H__
-
-#define CSR_PA(node, csr) ((csr) | ((uint64_t)(node) << 44))
-
-#define RST_BOOT 0x87E006001600ULL
-
-#endif
diff --git a/include/cavium/thunderx_fdt.h b/include/cavium/thunderx_fdt.h
deleted file mode 100644
index b25355ae3d..0000000000
--- a/include/cavium/thunderx_fdt.h
+++ /dev/null
@@ -1,11 +0,0 @@
-/**
- * (C) Copyright 2016, Cavium Inc.
- *
- * SPDX-License-Identifier:	GPL-2.0+
-**/
-#ifndef __THUNDERX_FDT_H__
-#define __THUNDERX_FDT_H__
-
-void thunderx_parse_bdk_config(void);
-
-#endif
diff --git a/include/cavium/thunderx_smi.h b/include/cavium/thunderx_smi.h
deleted file mode 100644
index a3929cab8f..0000000000
--- a/include/cavium/thunderx_smi.h
+++ /dev/null
@@ -1,12 +0,0 @@
-/**
- * (C) Copyright 2014, Cavium Inc.
- *
- * SPDX-License-Identifier:	GPL-2.0+
-**/
-
-#ifndef THUNDERX_SMI_H_
-#define THUNDERX_SMI_H_
-
-int thunderx_smi_initialize(bd_t *bis, unsigned int index);
-
-#endif /* THUNDERX_SMI_H_ */
diff --git a/include/cavium/thunderx_vnic.h b/include/cavium/thunderx_vnic.h
deleted file mode 100644
index e79284c8c5..0000000000
--- a/include/cavium/thunderx_vnic.h
+++ /dev/null
@@ -1,20 +0,0 @@
-/**
- * (C) Copyright 2014, Cavium Inc.
- *
- * SPDX-License-Identifier:	GPL-2.0+
-**/
-
-#ifndef VNIC_H_
-#define VNIC_H_
-
-struct nicpf;
-struct nicvf;
-
-int nicvf_initialize(struct udevice *dev, int vf_num);
-int thunderx_bgx_initialize(unsigned int bgx_idx, unsigned int node);
-void bgx_get_count(int node, int *bgx_count);
-int bgx_get_lmac_count(int node, int bgx_idx);
-void bgx_set_board_info(unsigned int bgx_id, unsigned int mdio_bus,
-			unsigned int *phy_addr);
-
-#endif /* VNIC_H_ */
diff --git a/include/configs/thunderx_81xx.h b/include/configs/thunderx_81xx.h
index 56910f6ae7..feb2e59ba7 100644
--- a/include/configs/thunderx_81xx.h
+++ b/include/configs/thunderx_81xx.h
@@ -11,11 +11,6 @@
 
 #define CONFIG_SPECIAL_SYNC_HANDLER
 
-/*#define CONFIG_ARMV8_SWITCH_TO_EL1*/
-//#define DEBUG
-
-#define CONFIG_SYS_64BIT
-
 /** Thunder 81xx does not support NOR flash */
 #define CONFIG_SYS_NO_FLASH
 
@@ -26,10 +21,6 @@
 /** Enable vsnprintf support */
 #define CONFIG_SYS_VSNPRINTF
 
-/** Identification string */
-#define CONFIG_IDENT_STRING	\
-	" for Cavium OcteonTX CN81XX ARM v8 Core"
-
 #define CONFIG_BOOTP_VCI_STRING		"Diagnostics"
 
 #define MEM_BASE			0x00500000
@@ -111,9 +102,6 @@
 
 /* Flat Device Tree Definitions */
 
-/** Enable flat device tree support */
-#define CONFIG_OF_LIBFDT
-
 /**
  * The board code has additional modification that it wants to make to the flat
  * device tree before handing it off to the Linux kernel.
@@ -146,19 +134,6 @@
  */
 #define CONFIG_SYS_MEM_TOP_HIDE		0x1000000
 
-/** Stack size for PSCI calls */
-#define CONFIG_AP_STACK_SIZE		65536
-
-/** Stack alignment for PSCI calls */
-#define CONFIG_AP_STACK_ALIGN		16
-
-/* PL011 Serial Configuration */
-/** Enable PL011 serial driver */
-#define CONFIG_PL01X_SERIAL
-
-/** PL011 serial clock speed */
-#define CONFIG_PL011_CLOCK		24000000
-
 /** Console serial port number */
 #define CONFIG_CONS_INDEX		1
 
@@ -168,12 +143,6 @@
 #define GICD_BASE			(0x801000000000)
 #define GICR_BASE			(0x801000002000)
 
-/** Base address of UART0 registers */
-#define CONFIG_SYS_SERIAL0		0x87e028000000
-
-/** Base address of UART1 registers */
-#define CONFIG_SYS_SERIAL1		0x87e029000000
-
 #define CONFIG_BAUDRATE			115200
 
 /* Net */
@@ -291,41 +260,16 @@
 #define CONFIG_PARTITION_UUIDS
 
 /* PCIe */
-/** Enable PCI support */
-#define CONFIG_PCI
-/** Enable 64-bit PCI addressing support */
-#define CONFIG_SYS_PCI_64BIT
 /** Show devices found on PCI bus */
 #define CONFIG_PCI_SCAN_SHOW
 #undef CONFIG_PCI_ENUM_ONLY
-/** Enable PCI plug-and-play support */
-#define CONFIG_PCI_PNP
-#define CONFIG_SYS_PCI_ARI
-/** Enable ThunderX ECAMS PCIE support */
-#define CONFIG_THUNDERX_ECAMS 1
-#define CONFIG_THUNDERX_RCS 3
-
-/** Define the cacheline size */
-#define CONFIG_SYS_CACHELINE_SIZE 128
 
 /** Number of I2C buses */
 #define CONFIG_SYS_MAX_I2C_BUS	2
 
-/** Enable ThunderX I2C support */
-#define CONFIG_SYS_I2C_THUNDERX
-
 /** I2C slave address */
 #define CONFIG_SYS_I2C_THUNDERX_SLAVE_ADDR	0x77
 
-/** I2C bus speed */
-#define CONFIG_SYS_I2C_SPEED			100000
-
-/** Speed for bus 0 */
-#define CONFIG_SYS_I2C_THUNDERX_SPEED_0		100000
-
-/** Speed for bus 1 */
-#define CONFIG_SYS_I2C_THUNDERX_SPEED_1		100000
-
 /** I2C bus real-time clock resides on */
 #define CONFIG_SYS_RTC_BUS_NUM 0
 
-- 
2.29.0

